spec/dynosaur/process/heroku_spec.rb

```diff
@@ -0,0 +1,55 @@
+describe Dynosaur::Process::Heroku do
+  describe '#start' do
+    subject { Dynosaur::Process::Heroku.new(task: task, args: args, opts: opts) }
+
+    let(:task) { 'fake:task' }
+    let(:args) { [] }
+    let(:opts) { {} }
+
+    let(:client) { instance_double(PlatformAPI::Client, dyno: dyno_accessor) }
+    let(:dyno_accessor) { instance_double(PlatformAPI::Dyno, create: api_response) }
+
+    let(:api_response) { { 'name' => 'run.9876' } }
+
+    before do
+      allow(PlatformAPI).to receive(:connect_oauth).with('dynosaur-test-key')
+        .and_return(client)
+    end
+
+    it 'invokes dyno.create with the rake task' do
+      command = 'rake fake:task --trace'
+      expect(dyno_accessor).to receive(:create)
+        .with('dynosaur-test-app', command: command, attach: false)
+        .and_return(api_response)
+      subject.start
+    end
+
+    it 'returns the dyno identifier' do
+      expect(subject.start).to eq('run.9876')
+    end
+
+    context 'when args are passed to the rake task' do
+      let(:args) { ['Hello, World', 1999] }
+
+      it 'adds the arguments to the rake task' do
+        command = 'rake fake:task["Hello, World",1999] --trace'
+        expect(dyno_accessor).to receive(:create)
+          .with('dynosaur-test-app', command: command, attach: false)
+          .and_return(api_response)
+        subject.start
+      end
+    end
+
+    context 'when dyno size is specified' do
+      let(:opts) { { size: 'big' } }
```

Comment:
My concern with doing that is as soon as Heroku changes its dyno pricing again, we'll have to update the gem. Then anyone who wants use the new dyno sizes will have to upgrade the gem, even if it mean jumping a whole version number. As it stands, passing in an invalid size will raise an error (via Heroku's Platform API).